File: print-template-frontend/src/api/print-task.ts
import request from '@/utils/request'

export interface PrintTask {
  id?: number
  taskNo?: string
  templateId: number
  templateName: string
  dataContent?: string
  printerName?: string
  paperSize?: string
  orientation?: string
  copies?: number
  duplex?: number
  status?: number
  failReason?: string
  printTime?: string
  createBy?: string
  createTime?: string
  updateTime?: string
}

export async function getPrintTasks(params: any) {
  return request({
    url: '/print-task/list',
    method: 'get',
    params
  })
}

export async function getPrintTask(id: number) {
  return request({
    url: `/print-task/${id}`,
    method: 'get'
  })
}

export async function createPrintTask(data: PrintTask) {
  return request({
    url: '/print-task',
    method: 'post',
    data
  })
}

export async function updatePrintTaskStatus(id: number, status: number, failReason?: string) {
  return request({
    url: `/print-task/${id}/status`,
    method: 'put',
    params: { status, failReason }
  })
}

export async function deletePrintTask(id: number) {
  return request({
    url: `/print-task/${id}`,
    method: 'delete'
  })
}

File: print-template-backend/src/main/java/com/printtemplate/module/print/entity/PrintTask.java
package com.printtemplate.module.print.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("sys_print_task")
public class PrintTask {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String taskNo;

    private Long templateId;

    private String templateName;

    @TableField(typeHandler = org.apache.ibatis.type.StringTypeHandler.class)
    private String dataContent;

    private String printerName;

    private String paperSize;

    private String orientation;

    private Integer copies;

    private Integer duplex;

    private Integer status;

    private String failReason;

    private LocalDateTime printTime;

    private String createBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}
